### PR TITLE
API 5.6

### PR DESCRIPTION
**Pull request by harshil21**
### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests
- [ ] Added myself alphabetically to `AUTHORS.rst` (optional)
- [ ] Added new classes & modules to the docs

### If the PR contains API changes (otherwise, you can delete this passage)

* New classes:
    - [ ] Added `self._id_attrs` and corresponding documentation
    - [ ] `__init__` accepts `**_kwargs`
    
* Added new shortcuts:
    - [x] In `Chat` & `User` for all methods that accept `chat/user_id`
    - [x] In `Message` for all methods that accept `chat_id` and `message_id`
    - [ ] For new `Message` shortcuts: Added `quote` argument if methods accepts `reply_to_message_id`
    - [x] In `CallbackQuery` for all methods that accept either `chat_id` and `message_id` or `inline_message_id`
    
* If relevant:

    - [x] Added new constants at `telegram.constants` and shortcuts to them as class variables
    - [ ] Added new handlers for new update types
    - [ ] Added new filters for new message (sub)types
    - [x] Added or updated documentation for the changed class(es) and/or method(s)
    - [x] Updated the Bot API version number in all places: `README.rst` and `README_RAW.rst` (including the badge), as well as `telegram.constants.BOT_API_VERSION`
    - [ ] Added logic for arbitrary callback data in `tg.ext.Bot` for new methods that either accept a `reply_markup` in some form or have a return type that is/contains `telegram.Message`

### CODE REVIEW

File: telegram/bot.py [hunk 1]
```diff
@@ -510,6 +515,9 @@ def send_message(
         if entities:
             data['entities'] = [me.to_dict() for me in entities]
 
+        if protect_content:
+            data['protect_content'] = protect_content
+
```
Comment (Bibo-Joshi) [inline telegram/bot.py:520]: I'd vote to just make this a parameter of `self._message` - I didn't check, but (almost) all methods that got this parameter now call `self._message` anyway and this saves us the `if protected_content:` :)

File: telegram/message.py [hunk 2]
```diff
@@ -2735,6 +2783,12 @@ def _parse_markdown(
                             'Strikethrough entities are not supported for Markdown ' 'version 1'
                         )
                     insert = '~' + text + '~'
+                elif entity.type == MessageEntity.SPOILER:
+                    if version == 1:
+                        raise ValueError(
+                            "Spoiler entities are not supported for Markdown " "version 1"
```
Comment (Bibo-Joshi) [inline telegram/message.py:2789]:
```suggestion
                            "Spoiler entities are not supported for Markdown version 1"
```

File: telegram/message.py [hunk 3]
```diff
@@ -2531,6 +2577,8 @@ def _parse_html(
                     insert = '<u>' + text + '</u>'
                 elif entity.type == MessageEntity.STRIKETHROUGH:
                     insert = '<s>' + text + '</s>'
+                elif entity.type == MessageEntity.SPOILER:
```
Comment (Bibo-Joshi) [inline telegram/message.py:2580]: these changes may be worth a `.. versionchanged::` in `text_html/markdown(_v2)`

**Review by Bibo-Joshi — COMMENTED** [review]
Thanks for the quick PR!

* which fixture did you fix ragarding message entities?
* I guess `test_bot` is yet to be adapted? :D

File: telegram/message.py (hunk 2 above)
Comment (harshil21) [reply telegram/message.py:2789]: black keeps doing this

**Comment by harshil21** [comment]
> Thanks for the quick PR!

Thanks for the quick review!

> * which fixture did you fix ragarding message entities?

See `tests/test_messageentity.py`. There was a spelling error.

> * I guess `test_bot` is yet to be adapted? :D

I'm not sure what more to change in it, i'll see

**Comment by Bibo-Joshi** [comment]
> > * which fixture did you fix ragarding message entities?
> 
> See `tests/test_messageentity.py`. There was a spelling error.

ah, you mean `type` -> `type_`? yeah, that's good

> > * I guess `test_bot` is yet to be adapted? :D
> 
> I'm not sure what more to change in it, i'll see

well, currently codecov complains about all the `if protected_content:` not being covered. While the coverage report will be more friendly when moving that clause into `self._message` (because then it's fewer of those), we should actually test it - should be doable by checking `Message.has_protected_content` on the return value in most cases.

File: telegram/message.py [hunk 4]
```diff
@@ -2634,6 +2634,9 @@ def text_html(self) -> str:
         Returns:
             :obj:`str`: Message text with entities formatted as HTML.
 
+        .. versionchanged:: 13.10
+           Spoiler entities are now formatted as HTML.
+
```
Comment (Bibo-Joshi) [inline telegram/message.py:2639]: please move up before the return value. same for the others

File: telegram/message.py [hunk 5]
```diff
@@ -2970,6 +2991,9 @@ def caption_markdown_v2_urled(self) -> str:
         Returns:
             :obj:`str`: Message caption with caption entities formatted as Markdown.
 
+        .. versionchanged:: 13.10
+           Spoiler entities are now formatted as Markdown V2.
+
```
Comment (Bibo-Joshi) [inline telegram/message.py:2996]: maybe also a `.. versionchanged::` for `text/caption_markdown` along the lines "raises ValueError if the message contains spoiler entities"? and a `Raises:` block, if it doesn't have that yet …

**Review by Bibo-Joshi — COMMENTED** [review]
Nice! Apart from nitpicking, I guess it's ready, though I'd like for @Poolitzer to have a look as well. 

just double checking: are you both comfortable with postponing `Defaults.protect_content` to v14? If you prefer to include it directly, we can do that.

**Review by Poolitzer — APPROVED** [review]
LGTM!

**Comment by Kazama009** [comment]
pain in the ass -_-

**Comment by Bibo-Joshi** [comment]
what?

**Pull request merged**